pkg/compute/management_client.go

```diff
@@ -68,17 +65,8 @@ func (m *ManagementClient) getNodeInfo(ctx context.Context) models.NodeInfo {
 	return ni
 }
 
-// RegisterNode sends a registration request to the requester node. If we successfully
-// register, a sentinel file is created to indicate that we are registered. If present
-// the requester node will know it is already registered.  If not present, it will
-// attempt to register again, expecting the requester node to gracefully handle any
-// previous registrations.
+// RegisterNode sends a registration request to the requester node.
 func (m *ManagementClient) RegisterNode(ctx context.Context) error {
-	if m.registrationFile.Exists() {
-		log.Ctx(ctx).Debug().Msg("not registering with requester, already registered")
-		return nil
-	}
-
```

Comment:
Why are we removing the registration state? This means the compute node will have to re-register every time at startup and the node will fail to start if it is offline or disconnected from the requester.

Overall, I think we need to do better for node registration and allow the compute node to periodically re-check with the requester if the registration info match, and re-register if needed. So not only during startup, and should be best effort if the compute node is offline. Meaning the compute node should continue to run the locally allocated jobs even if it fails to register or re-register. I have a NodeManagerV2 placeholder for 1.5 that should track and fix most issues related to node management.

Since we don't have full offline capabilities today, it might make sense to force compute nodes to re-register at startup if we know what we are doing, and we can fix things with NodeManagerV2

Reply:
> Why are we removing the registration state?

Its a requirement for the migration added here to work since we are deleting the requester nodes nodestore. Removing this lock ensures each time a compute node is started it attempts to register itself wit the requester. This is important since in the event a requester loses it nodestate store (i.e. the migration) compute nodes will re-register themselves with it. If they have already registered with a requester node registering again idempotent.

> the node will fail to start if it is offline or disconnected from the requester

True, although the compute nodes will also fail to start if starting fresh and they are unable to connect to the requester. I don't think a failure here should result in an aborted start, rather the node should continue attempting to register itself until successful (e.g. loop until success)

Reply:
Yeah I understand. As I mentioned it is fine to remove the registration state for now